service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateService.kt

```diff
@@ -120,7 +120,15 @@ fun updateVardaChild(db: Database.Connection, client: VardaClient, serviceNeedDi
 
         if (!db.read { it.hasVardaServiceNeeds(evakaChildId) }) {
             logger.info("VardaUpdate: deleting child for not having service needs ($evakaChildId)")
-            deleteChildDataFromVardaAndDb(db, client, evakaChildId.raw)
+
+            if (!deleteChildDataFromVardaAndDb(db, client, evakaChildId.raw)) {
+                // If child has 0 service needs in varda, and delete fails, we need to retry it via the reset mechanism
+                db.transaction {
+                    it.createUpdate("UPDATE varda_reset_child SET reset_timestamp = null WHERE evaka_child_id = :id")
+                        .bind("id", evakaChildId)
+                        .execute()
```

Comment:
This is the only SQL in this file. Should it be moved to `VardaQueries.kt` or somewhere? 🤔